msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/ForumsEmailService.java

```diff
@@ -345,4 +348,30 @@ private boolean isValidEmail(String email) {
 		
 	}
 
+	private String getAnonAwareAuthor(Message message)
+	{
+		if (isUseAnonymousId(message.getTopic()))
+		{
+			AnonymousManager anonMan = (AnonymousManager)ComponentManager.get("org.sakaiproject.api.app.messageforums.AnonymousManager");
+			String siteId = ToolManager.getCurrentPlacement().getContext();
+			String userId = message.getAuthorId();
+			return anonMan.getOrCreateAnonId(siteId, userId);
+		}
+		return message.getAuthor();
+	}
+
+	private boolean isUseAnonymousId(Topic topic)
+	{
+		if (topic == null)
+		{
+			throw new IllegalArgumentException("isUseAnonymousId invoked with null topic");
+		}
+
+		/* 
+		 * TODO: we ignore isRevealIDsToRoles / isIdentifyAnonAuthors here because everybody's getting the same email.
+		 * This is okay, but if isRevealIDsToRoles is enabled, it would be better to send two separate emails (one revealed, one anonymous) 
+		 * so that the instructor can't match the message up with the UI to discover the anonymous mapping for the author
+		 */
+		return ServerConfigurationService.getBoolean("msgcntr.forums.anonymous.enable", true) && topic.getPostAnonymous();
```

Comment:
I would put this sakai.property name and default into a static final variable on the class, or in a common place for use. Same with the other place(s) you're checking this property.